lib/rucio/tests/temp_test_env.py

```diff
@@ -0,0 +1,205 @@
+        if not self._client:
+            self._client = Client()
+        return self._client
+
+    @property
+    def upload_client(self):
+        if not self._upload_client:
+            self._upload_client = UploadClient(self.client)
+        return self._upload_client
+
+    @transactional_session
+    def cleanup(self, session=None):
+        if not self.created_dids:
+            return
+
+        # Cleanup Transfers
+        session.query(models.Source).filter(or_(and_(models.Source.scope == did['scope'],
+                                                     models.Source.name == did['name'])
+                                                for did in self.created_dids)).delete(synchronize_session=False)
+        session.query(models.Request).filter(or_(and_(models.Request.scope == did['scope'],
+                                                      models.Request.name == did['name'])
+                                                 for did in self.created_dids)).delete(synchronize_session=False)
+
+        # Cleanup Locks Rules
+        query = session.query(models.ReplicationRule.id).filter(or_(and_(models.ReplicationRule.scope == did['scope'],
+                                                                         models.ReplicationRule.name == did['name'])
+                                                                    for did in self.created_dids))
+        for rule_id, in query:
+            rule_core.delete_rule(rule_id, session=session)
+
+        # Cleanup Replicas and Parent Datasets
+        dids_by_rse = {}
+        replicas = list(replica_core.list_replicas(self.created_dids, all_states=True, session=session))
+        for replica in replicas:
+            for rse_id in replica['rses']:
+                dids_by_rse.setdefault(rse_id, []).append({'scope': replica['scope'], 'name': replica['name']})
+        for rse_id, dids in dids_by_rse.items():
+            replica_core.delete_replicas(rse_id=rse_id, files=dids, session=session)
+
+    def upload_test_file(self, rse_name, scope=None, name=None, path=None, return_full_item=False):
+        if not scope:
+            scope = self.default_scope
+        elif isinstance(scope, str):
+            scope = InternalScope(scope, vo=self.vo)
+        if not path:
+            path = file_generator()
+        if not name:
+            name = os.path.basename(path)
+        item = {
+            'path': path,
+            'rse': rse_name,
+            'did_scope': str(scope),
+            'did_name': name,
+            'guid': generate_uuid(),
+        }
+        self.upload_client.upload([item])
+        did = {'scope': scope, 'name': name}
+        self.created_dids.append(did)
+        return item if return_full_item else did
+
+
+class TemporaryTestEnv(TemporaryRSEFactory, TemporaryFileFactory):
+    def __init__(self, vo, default_scope):
+        TemporaryRSEFactory.__init__(self, vo=vo)
+        TemporaryFileFactory.__init__(self, vo=vo, default_scope=default_scope)
```

Comment:
why not use `super(…)` here and in the cleanup? Also I think a fixture for each that just looks like this would be a better fit and seems more modular:
```python3
@pytest.fixture
def rse_factory(vo):
    factory = TemporaryRSEFactory(vo=vo)
    yield factory
    factory.cleanup()
```

Reply:
You are right, I started by having a single class, then split it into 2. It's better to have separate fixtures. Split it and amended the commits